core-bundle/src/Security/Voter/DataContainer/PageTypeAccessVoter.php

```diff
@@ -90,10 +90,12 @@ private function validateFirstLevelType(CreateAction|DeleteAction|ReadAction|Upd
 
         $type = $action->getNew()['type'] ?? ($action instanceof UpdateAction ? $action->getCurrent()['type'] : null);
         $pid = (int) ($action->getNewPid() ?? ($action instanceof UpdateAction ? $action->getCurrentPid() : null));
+        $currentPid = $action instanceof UpdateAction ? (int) $action->getCurrentPid() : null;
```

Comment:
```suggestion
        $currentPid = $action instanceof UpdateAction ? (int) $action->getCurrentPid() : null;
        $pid = (int) ($action->getNewPid() ?? $currentPid);
```

This would save one `$action->getCurrentPid()` call.